### ISSUE TITLE
Weird High DPI behaviour on Windows 10

### ISSUE THREAD

**Issue by johnnovak** [dosbox-staging/dosbox-staging#2135]
Okay @GranMinigun, I'm a bit late to the party... I gave the Windows High DPI support a go today, and I'm seeing some weird behaviour. In short, it seems that when using 125%, 150%, etc. scaling in Windows, the framebuffer is still rendered at the unscaled (100%) resolution, then Windows simply stretches that to whatever scaling factor you've set.

Now, I know you guys did test this extensively, but is there a chance you haven't noticed the scaling artifacts perhaps in Windows? Also, this High DPI works perfectly fine on macOS. It's harder to notice with 256-colour 320x200 VGA games that have "busy" graphics, but quite easy to spot with EGA games that feature checkerboard dither patterns. Moreover, when I took some post-shader screenshots by pressing Alt+F5, the resulting image always has the unscaled 100% dimensions, so that's the ultimate proof.

The game Windwalker has large areas of checkerboard dither in the intro screen, so it's best to test with that. Just unzip and start with `-noprimaryconf` (shader is included, but you could actually just test with `sharp` as well).

[Windwalker.zip](https://github.com/dosbox-staging/dosbox-staging/files/10250540/Windwalker.zip)

I took some screenshots in windowed mode because when I try to take OS-level screenshots in fullscreen, I'm only getting a black screen.

## 100%

This is how things should look like; everything is perfect at 100% scaling. Notice the dark-grey dither pattern is completely even and smooth, there's zero moire.

![windwalker-100](https://user-images.githubusercontent.com/698770/208218287-bf7037b6-e4c3-457f-9f87-70e5d0a3384f.png)

## 125%

At 125% scaling there's tons of scaling artifacts which are very noticeable in the dark-grey dither pattern (there's moire and a general blurriness too). Clearly, the image is rendered at 100% first, then Windows just stretches the resulting image to 125%. My understanding is that the whole point of the feature is that we would actually render at 125%, completely bypassing the OS-level scaling, right? (must be, because that's how High DPI behaves on macOS, and otherwise there's not much point in the feature) At 150% the scaling artifacts are even more pronounced (uneven scanlines, etc.)

<img alt="windwalker-125" src="https://user-images.githubusercontent.com/698770/208218296-415250c7-1d7c-47c5-b98d-abe595a2c764.png">

**Comment by GranMinigun** [comment]
And with texture output? That's the only one I was able to test.

**Comment by GranMinigun** [comment]
On a second thought, can you please check whether it looks identical without shaders, with `openglnb` output, with the exact same window pixel dimensions? Something like:
```
[render]
glshader = none

[sdl]
output = openglnb
viewport_resolution = fit
windowresolution = 800x600 ; w/ 100% scaling, and
windowresolution = 640x480 ; w/ 125% scaling
```

**Comment by johnnovak** [comment]
Tested as the above config, and the results are in line with my previous findings: internally, the specified window/viewport resolution is used (as if 100% scaling was used), then Windows stretches the result to the scaled resolution using some really shitty bilinear interpolation algorithm 😐 

So, I don't think High DPI on Windows is ready for prime time.

## windowresolution = 800x600 ; w/ 100% scaling

### OS-level capture (Alt-PrintScreen)

![windwalker-800x600-x100](https://user-images.githubusercontent.com/698770/208230073-fddfed1c-ea54-4450-add6-eff0ce6093ba.png)

### Alt-F5 capture (resulting image is 800x600)

This is pixel-identical to the OS-level capture, as expected

![capture-800x600](https://user-images.githubusercontent.com/698770/208230055-8ada6448-e2c7-42be-93b6-31985e58b129.png)


## windowresolution = 640x480 ; w/ 125% scaling

### OS-level capture (Alt-PrintScreen)
Zoom in and you'll see the really crappy linear interpolation applied on top of the framebuffer...

![windwalker-640x480-x125](https://user-images.githubusercontent.com/698770/208230078-2471d8a8-615e-4228-b719-e791cf11750f.png)


### Alt-F5 capture (resulting image is 640x480)

*Not* identical to the OS-level capture

![capture-640x480](https://user-images.githubusercontent.com/698770/208230049-ab0b8d4c-5dfd-4a72-bb8c-7782d6973c36.png)

**Comment by GranMinigun** [comment]
Can you also compare with `texturenb` output, please? Maybe also with different `texture_renderer` values, like `opengl` vs `direct3d`. And just in case, check if Windows overrides scaling setting (right-click on dosbox.exe => Properties => Compatibility tab => Change high DPI settings => boxes should be unticked). Backbuffer size should be handled for us by SDL, as far as I'm aware. What happens if you resize the window? Switch to fullscreen and back?

**Comment by johnnovak** [comment]
I will test all this, but I think only tomorrow.

**Comment by johnnovak** [comment]
> And just in case, check if Windows overrides scaling setting (right-click on dosbox.exe => Properties => Compatibility tab => Change high DPI settings => boxes should be unticked).

I've checked, and both checkboxes are unticked.

What I've done is run the game in fullscreen with the following settings, then create a screenshot with Alt-F5.

```
[sdl]
output = texturenb
viewport_resolution = fit
```

At 100x scaling the resulting image is 1920x1080 as expected.

![texturenb-100x](https://user-images.githubusercontent.com/698770/208286326-a5ce2b15-9b6a-42c3-bfa8-9144565fdb23.png)

At 125x scaling it's only 1536x864, so again this is the proof that we're rendering at a smaller size, then Windows or SDL just stretches the image to 1920x1080.

This stretched image fill the screen with 125x enabled, of course. Switching between fullscreen & windowed, or resizing the window doesn't change anything. There is a slight difference between the `direct3d` and `opengl` texture renderers: with `direct3d`, I can see some linear interpolation artifacts on the stretched image (it's very similar to our "bilinear sharp"), while with the `opengl` renderer the streching is done via nearest-neighbour (and looks terrible).

![texturenb-125x](https://user-images.githubusercontent.com/698770/208286328-481dc90a-00f4-467d-83b7-d13185f7c3d4.png)

So, to sum up, I can't make DOSBox render at the "target resolution"; it always renders at a smaller resolutions, which then gets stretched by the specified scaling factor (e.g. 125%) to the actual target resolution. This is with all `output` and `texture_renderer` modes I tried.

My conclusion: hi-dpi support does not work on Windows (at least on my machine).

**Comment by GranMinigun** [comment]
This is very weird and I'm not sure where to dig, because I don't see any system-performed bitmap stretching in my Win11 VM:

![dosbox_000](https://user-images.githubusercontent.com/11149380/208294711-52b86157-c2ef-49bf-942b-7712cd351c48.png)
![mm3_000](https://user-images.githubusercontent.com/11149380/208294715-159dcd6f-8456-4d00-a1c8-eae6ae7b60aa.png)

Captured with Alt+F5, 125% scaling set in the system. These are native res, as you can see (and, weirdly, with transparency on the right side). Could be driver-related? The DPI awareness flag somehow fails (it should print a warning)? I assume you're on Windows 10, 1703 or later. Absolutely no idea. At this point, it's probably a better idea to ask SDL folks what's going on, or at least get one more confirmation on a different machine.

**Comment by johnnovak** [comment]
Yeah, I'm on the latest Win10 and I'm using an RTX 3060. Well, things can be different in a VM. Ideally, you'd really want to work on this on a real WIndows box, otherwise it can be quite difficult to troubleshoot these issues (but thanks for trying to get it to work, hey!)

I'm not interested personally to take on the high-DPI work, so feel free to raise it with the SDL folks.

**Comment by ericwa** [comment]
Hi, I worked on the SDL2 windows highdpi support, and am trying to repro this issue.

On my Win11 system it seems to be working as expected.

- I grabbed the `dosbox-staging-windows-msys2-x86_64-v0.81.0-alpha-29-gd81eb ` build from https://github.com/dosbox-staging/dosbox-staging/actions/runs/3772871956
- my Windows scaling is set to 125%
- Not a VM. Windows 11 22H2
- NVIDIA 1070 Ti, driver 527.56
- I tried the dosbox-staging settings from https://github.com/dosbox-staging/dosbox-staging/issues/2135#issuecomment-1356077290

This all looks correct from what I can see.

![image](https://user-images.githubusercontent.com/239161/209452622-6ed6946b-fc5a-40f2-a76b-b1182c204ef4.png)

Full log output:

```
date       time         |
2022-12-24 15:42:37.735 | arguments: C:\\Users\\Eric\\Downloads\\dosbox-staging-windows-msys2-x86_64-v0.81.0-alpha-29-gd81eb\\dosbox.exe
2022-12-24 15:42:37.735 | Current dir: C:\Users\Eric\Downloads\dosbox-staging-windows-msys2-x86_64-v0.81.0-alpha-29-gd81eb
2022-12-24 15:42:37.735 | stderr verbosity: 0
2022-12-24 15:42:37.735 | -----------------------------------
2022-12-24 15:42:37.735 | dosbox-staging version 0.81.0-alpha-29-gd81eb96cf
2022-12-24 15:42:37.735 | ---
2022-12-24 15:42:37.735 | LOG: Loguru version 2.1.0 initialized
2022-12-24 15:42:37.749 | SDL: version 2.26.1 initialized (windows video and wasapi audio)
2022-12-24 15:42:37.752 | CONFIG: Loaded primary conf file C:\Users\Eric\AppData\Local\DOSBox\dosbox-staging.conf
2022-12-24 15:42:37.752 | LANG: Using internal English language messages
2022-12-24 15:42:37.756 | RENDER: Using GLSL shader 'none'
2022-12-24 15:42:37.757 | DISPLAY: Initialized 640x480 window-mode using Nearest-neighbour scaling on 1728p display-0
2022-12-24 15:42:37.935 | OPENGL: Vendor: NVIDIA Corporation
2022-12-24 15:42:37.935 | OPENGL: Version: 4.6.0 NVIDIA 527.56
2022-12-24 15:42:37.935 | OPENGL: GLSL version: 4.60 NVIDIA
2022-12-24 15:42:37.935 | OPENGL: Pixel buffer object: available
2022-12-24 15:42:37.935 | OPENGL: NPOT textures: supported
2022-12-24 15:42:37.937 | VIDEO: Using the DOS video mode's frame rate
2022-12-24 15:42:37.941 | MEMORY: Using 4096 DOS memory pages (16 MiB) at address: 0000011100CCB040
2022-12-24 15:42:37.942 | VIDEO: Initialized S3 Trio64 VESA 2.0 compatible with 4-MiB of FP DRAM supporting 86 modes
2022-12-24 15:42:37.956 | MIXER: Negotiated 2-channel 48000-Hz audio in 1024-frame blocks
2022-12-24 15:42:37.956 | MIXER: Reverb disabled
2022-12-24 15:42:37.956 | MIXER: Chorus disabled
2022-12-24 15:42:37.956 | MIXER: Master compressor enabled
2022-12-24 15:42:38.063 | MIDI: Opened device: win32
2022-12-24 15:42:38.065 | OPL: Operating at 48000 Hz without resampling
2022-12-24 15:42:38.065 | OPL: Running OPL3 on ports 220h and 388h
2022-12-24 15:42:38.065 | SB16: Sound Blaster 16 OPL output filter enabled
2022-12-24 15:42:38.065 | SB: Operating at 22050 Hz and upsampling to the output rate
2022-12-24 15:42:38.065 | SB16: Modern DAC output filter enabled
2022-12-24 15:42:38.065 | SB16: BLASTER=A220 I7 D1 H5 T6
2022-12-24 15:42:38.065 | SB16: Running on port 220h, IRQ 7, DMA 1, and high DMA 5
2022-12-24 15:42:38.065 | PCSPEAKER: Operating at 48000 Hz without resampling
2022-12-24 15:42:38.065 | PCSPEAKER: Initialized discrete model
2022-12-24 15:42:38.065 | PCSPEAKER: Highpass filter enabled (18 dB/oct at 120 Hz)
2022-12-24 15:42:38.065 | PCSPEAKER: Lowpass filter enabled (12 dB/oct at 4800 Hz)
2022-12-24 15:42:38.065 | MOUSE: Will be captured immediately on start
2022-12-24 15:42:38.066 | MOUSE: Middle button will capture/release the mouse (clicks not sent to the game/program)
2022-12-24 15:42:38.066 | MOUSE (PS/2): Standard, 3 buttons
2022-12-24 15:42:38.071 | SLIRP: Slirp version: 4.7.0
2022-12-24 15:42:38.071 | SLIRP: Successfully initialized
2022-12-24 15:42:38.072 | NE2000: Base=0x300 irq=3
2022-12-24 15:42:38.073 | CONFIG: Loaded drive conf file resources\drives\y.conf
2022-12-24 15:42:38.200 | MAPPER: no joysticks found
2022-12-24 15:42:38.200 | MAPPER: Loaded default key bindings
2022-12-24 15:42:38.204 | SHELL: Redirect output to NUL
2022-12-24 15:42:38.205 | MOUNT: Path 'resources\drives\y' found
2022-12-24 15:42:38.210 | SDL: Using standard SDI (auto) display refresh rate of 120 Hz
2022-12-24 15:42:38.248 | DISPLAY: Text 640x400 (3h) at 70.087 Hz VFR, scaled by 1.3x1.5 to 800x600 with 1.2 pixel-aspect
```

**Comment by johnnovak** [comment]
> This all looks correct from what I can see.

Please perform the exact steps I described in my first post, with the exact same game (ZIP file included), then report back.

Things can "look alright", but you need to perform those steps to the letter, and create screenshots that will actually reveal what the exact dimensions or the framebuffer are, as I explained.

**Comment by ericwa** [comment]
Sure. I extracted a fresh copy of `dosbox-staging-windows-msys2-x86_64-v0.81.0-alpha-29-gd81eb`, extracted the `drives`, `glshaders`, and `dosbox.conf` from Windwalker.zip into my `dosbox-staging-windows-msys2-x86_64-v0.81.0-alpha-29-gd81eb` folder, and ran `dosbox.exe -noprimaryconf`

Screenshot (with Windows snipping tool):

![image](https://user-images.githubusercontent.com/239161/209454435-1d46f368-ed73-485e-8a1d-9de742a182c8.png)

Alt+F5 isn't generating a screenshot or at least I can't find it.

Log output:

```
2022-12-24 18:07:50.942 | arguments: C:\\Users\\Eric\\Downloads\\dosbox-staging-windows-msys2-x86_64-v0.81.0-alpha-29-gd81eb\\dosbox.exe -noprimaryconf
2022-12-24 18:07:50.942 | Current dir: C:\Users\Eric\Downloads\dosbox-staging-windows-msys2-x86_64-v0.81.0-alpha-29-gd81eb
2022-12-24 18:07:50.942 | stderr verbosity: 0
2022-12-24 18:07:50.942 | -----------------------------------
2022-12-24 18:07:50.942 | dosbox-staging version 0.81.0-alpha-29-gd81eb96cf
2022-12-24 18:07:50.942 | ---
2022-12-24 18:07:50.942 | LOG: Loguru version 2.1.0 initialized
2022-12-24 18:07:50.959 | SDL: version 2.26.1 initialized (windows video and wasapi audio)
2022-12-24 18:07:50.961 | CONFIG: Loaded local conf file dosbox.conf
2022-12-24 18:07:50.961 | LANG: Using internal English language messages
2022-12-24 18:07:50.962 | RENDER: Shader file 'crt-aperture.ega-lores' not found
2022-12-24 18:07:50.963 | RENDER:
2022-12-24 18:07:50.963 | RENDER: List of available GLSL shaders
2022-12-24 18:07:50.963 | RENDER: ------------------------------
2022-12-24 18:07:50.963 | RENDER: Path '.\glshaders' has:
2022-12-24 18:07:50.963 | RENDER:         `- crt-hyllian-updated.ega-lores
2022-12-24 18:07:50.963 | RENDER:
2022-12-24 18:07:50.963 | RENDER: Path 'resources\glshaders' has:
2022-12-24 18:07:50.963 | RENDER:         |- crt\aperture
2022-12-24 18:07:50.963 | RENDER:         |- crt\caligari
2022-12-24 18:07:50.963 | RENDER:         |- crt\easymode
2022-12-24 18:07:50.963 | RENDER:         |- crt\easymode.tweaked
2022-12-24 18:07:50.963 | RENDER:         |- crt\fakelottes
2022-12-24 18:07:50.963 | RENDER:         |- crt\fakelottes.tweaked
2022-12-24 18:07:50.963 | RENDER:         |- crt\geom
2022-12-24 18:07:50.963 | RENDER:         |- crt\geom.tweaked
2022-12-24 18:07:50.963 | RENDER:         |- crt\hyllian-updated
2022-12-24 18:07:50.963 | RENDER:         |- crt\hyllian
2022-12-24 18:07:50.963 | RENDER:         |- crt\lottes-fast
2022-12-24 18:07:50.963 | RENDER:         |- crt\lottes-fast.subtle+gain
2022-12-24 18:07:50.963 | RENDER:         |- crt\lottes
2022-12-24 18:07:50.963 | RENDER:         |- crt\lottes.tweaked
2022-12-24 18:07:50.963 | RENDER:         |- crt\mattias
2022-12-24 18:07:50.963 | RENDER:         |- crt\pi-vertical
2022-12-24 18:07:50.963 | RENDER:         |- crt\pi
2022-12-24 18:07:50.963 | RENDER:         |- crt\yee64
2022-12-24 18:07:50.963 | RENDER:         |- crt\yeetron
2022-12-24 18:07:50.963 | RENDER:         |- crt\zfast-composite
2022-12-24 18:07:50.963 | RENDER:         |- crt\zfast
2022-12-24 18:07:50.963 | RENDER:         |- interpolation\catmull-rom
2022-12-24 18:07:50.963 | RENDER:         |- interpolation\sharp
2022-12-24 18:07:50.963 | RENDER:         |- none
2022-12-24 18:07:50.963 | RENDER:         |- scaler\advinterp2x
2022-12-24 18:07:50.963 | RENDER:         |- scaler\advinterp3x
2022-12-24 18:07:50.963 | RENDER:         |- scaler\advmame2x
2022-12-24 18:07:50.963 | RENDER:         |- scaler\advmame3x
2022-12-24 18:07:50.963 | RENDER:         |- scaler\rgb2x
2022-12-24 18:07:50.963 | RENDER:         |- scaler\rgb3x
2022-12-24 18:07:50.964 | RENDER:         |- scaler\scan2x
2022-12-24 18:07:50.964 | RENDER:         |- scaler\scan3x
2022-12-24 18:07:50.964 | RENDER:         |- scaler\tv2x
2022-12-24 18:07:50.964 | RENDER:         `- scaler\tv3x
2022-12-24 18:07:50.964 | RENDER:
2022-12-24 18:07:50.964 | RENDER: Path 'C:\Users\Eric\AppData\Local\DOSBox\glshaders' has no shaders
2022-12-24 18:07:50.964 | RENDER:
2022-12-24 18:07:50.964 | RENDER: Path 'C:\Users\Eric\Downloads\dosbox-staging-windows-msys2-x86_64-v0.81.0-alpha-29-gd81eb\resources\glshaders' has:
2022-12-24 18:07:50.964 | RENDER:         |- crt\aperture
2022-12-24 18:07:50.964 | RENDER:         |- crt\caligari
2022-12-24 18:07:50.964 | RENDER:         |- crt\easymode
2022-12-24 18:07:50.964 | RENDER:         |- crt\easymode.tweaked
2022-12-24 18:07:50.964 | RENDER:         |- crt\fakelottes
2022-12-24 18:07:50.964 | RENDER:         |- crt\fakelottes.tweaked
2022-12-24 18:07:50.964 | RENDER:         |- crt\geom
2022-12-24 18:07:50.964 | RENDER:         |- crt\geom.tweaked
2022-12-24 18:07:50.964 | RENDER:         |- crt\hyllian-updated
2022-12-24 18:07:50.964 | RENDER:         |- crt\hyllian
2022-12-24 18:07:50.964 | RENDER:         |- crt\lottes-fast
2022-12-24 18:07:50.964 | RENDER:         |- crt\lottes-fast.subtle+gain
2022-12-24 18:07:50.964 | RENDER:         |- crt\lottes
2022-12-24 18:07:50.964 | RENDER:         |- crt\lottes.tweaked
2022-12-24 18:07:50.964 | RENDER:         |- crt\mattias
2022-12-24 18:07:50.964 | RENDER:         |- crt\pi-vertical
2022-12-24 18:07:50.964 | RENDER:         |- crt\pi
2022-12-24 18:07:50.964 | RENDER:         |- crt\yee64
2022-12-24 18:07:50.964 | RENDER:         |- crt\yeetron
2022-12-24 18:07:50.964 | RENDER:         |- crt\zfast-composite
2022-12-24 18:07:50.964 | RENDER:         |- crt\zfast
2022-12-24 18:07:50.964 | RENDER:         |- interpolation\catmull-rom
2022-12-24 18:07:50.964 | RENDER:         |- interpolation\sharp
2022-12-24 18:07:50.964 | RENDER:         |- none
2022-12-24 18:07:50.964 | RENDER:         |- scaler\advinterp2x
2022-12-24 18:07:50.964 | RENDER:         |- scaler\advinterp3x
2022-12-24 18:07:50.964 | RENDER:         |- scaler\advmame2x
2022-12-24 18:07:50.964 | RENDER:         |- scaler\advmame3x
2022-12-24 18:07:50.964 | RENDER:         |- scaler\rgb2x
2022-12-24 18:07:50.964 | RENDER:         |- scaler\rgb3x
2022-12-24 18:07:50.964 | RENDER:         |- scaler\scan2x
2022-12-24 18:07:50.964 | RENDER:         |- scaler\scan3x
2022-12-24 18:07:50.964 | RENDER:         |- scaler\tv2x
2022-12-24 18:07:50.964 | RENDER:         `- scaler\tv3x
2022-12-24 18:07:50.964 | RENDER:
2022-12-24 18:07:50.964 | RENDER: The above shaders can be used exactly as listed in the "glshader"
2022-12-24 18:07:50.964 | RENDER: conf setting, without the need for the resource path or .glsl extension.
2022-12-24 18:07:50.964 | RENDER:
2022-12-24 18:07:50.965 | RENDER: Using GLSL shader 'none'
2022-12-24 18:07:50.966 | DISPLAY: Limiting viewport resolution to 1000x750
2022-12-24 18:07:50.966 | DISPLAY: Initialized 1706x1279 window-mode using Bilinear scaling on 1728p display-0
2022-12-24 18:07:51.121 | OPENGL: Vendor: NVIDIA Corporation
2022-12-24 18:07:51.121 | OPENGL: Version: 4.6.0 NVIDIA 527.56
2022-12-24 18:07:51.121 | OPENGL: GLSL version: 4.60 NVIDIA
2022-12-24 18:07:51.121 | OPENGL: Pixel buffer object: available
2022-12-24 18:07:51.121 | OPENGL: NPOT textures: supported
2022-12-24 18:07:51.122 | VIDEO: Using the DOS video mode's frame rate
2022-12-24 18:07:51.126 | MEMORY: Using 4096 DOS memory pages (16 MiB) at address: 000001E621100040
2022-12-24 18:07:51.142 | MIXER: Negotiated 2-channel 48000-Hz audio in 1024-frame blocks
2022-12-24 18:07:51.142 | MIXER: Reverb disabled
2022-12-24 18:07:51.142 | MIXER: Chorus disabled
2022-12-24 18:07:51.142 | MIXER: Master compressor enabled
2022-12-24 18:07:51.242 | MIDI: Opened device: win32
2022-12-24 18:07:51.243 | OPL: Operating at 48000 Hz without resampling
2022-12-24 18:07:51.244 | OPL: Running OPL3 on ports 220h and 388h
2022-12-24 18:07:51.244 | SB16: Sound Blaster 16 OPL output filter enabled
2022-12-24 18:07:51.244 | SB: Operating at 22050 Hz and upsampling to the output rate
2022-12-24 18:07:51.244 | SB16: Modern DAC output filter enabled
2022-12-24 18:07:51.244 | SB16: BLASTER=A220 I7 D1 H5 T6
2022-12-24 18:07:51.244 | SB16: Running on port 220h, IRQ 7, DMA 1, and high DMA 5
2022-12-24 18:07:51.536 | INNOVATION: Operating at 48000 Hz without resampling
2022-12-24 18:07:51.537 | INNOVATION: Running on port 280h with a SID 8580 at 0.895 MHz filtering at 50%
2022-12-24 18:07:51.537 | PCSPEAKER: Operating at 48000 Hz without resampling
2022-12-24 18:07:51.537 | PCSPEAKER: Initialized discrete model
2022-12-24 18:07:51.537 | PCSPEAKER: Highpass filter enabled (18 dB/oct at 120 Hz)
2022-12-24 18:07:51.537 | PCSPEAKER: Lowpass filter enabled (12 dB/oct at 4800 Hz)
2022-12-24 18:07:51.538 | MOUSE: Will be captured immediately on start
2022-12-24 18:07:51.538 | MOUSE: Middle button will capture/release the mouse (clicks not sent to the game/program)
2022-12-24 18:07:51.538 | MOUSE (PS/2): Standard, 3 buttons
2022-12-24 18:07:51.555 | SLIRP: Slirp version: 4.7.0
2022-12-24 18:07:51.555 | SLIRP: Successfully initialized
2022-12-24 18:07:51.555 | NE2000: Base=0x300 irq=3
2022-12-24 18:07:51.558 | CONFIG: Loaded drive conf file resources\drives\y.conf
2022-12-24 18:07:51.558 | AUTOEXEC: Using autoexec from one or more joined sections
2022-12-24 18:07:51.682 | MAPPER: no joysticks found
2022-12-24 18:07:51.682 | MAPPER: Loaded default key bindings
2022-12-24 18:07:51.696 | MOUNT: Path 'drives\c' found
2022-12-24 18:07:51.697 | SHELL: Redirect output to NUL
2022-12-24 18:07:51.698 | MOUNT: Path 'resources\drives\y' found
2022-12-24 18:07:51.702 | SDL: Using standard SDI (auto) display refresh rate of 120 Hz
2022-12-24 18:07:51.710 | DISPLAY: Text 640x350 (3h) at 59.702 Hz VFR, scaled by 2.0x2.7 to 1250x938 with 1.4 pixel-aspect
2022-12-24 18:07:51.743 | DISPLAY: EGA 320x200 16 color (Dh) at 59.923 Hz VFR, scaled by 3.9x4.7 to 1250x938 with 1.2 pixel-aspect
```

Can you post your log output?

**Comment by johnnovak** [comment]
Okay @ericwa and @GranMinigun, I've retested this again, and it looks like this was either a false alarm, or I did my above testing with an earlier version that indeed had the problem (sadly, I did not record the exact version number, and I am in no mood to bisect it...) Or some obscure Windows problem that got fixed by a reboot 🤷🏻 In any case, apologies for the confusion, and here's the proof that it's working as expected.

Tested it with these builds:

- `dosbox-staging version 0.80.0-2-gdcb440822` (that's the last 0.80.0 release)
- `dosbox-staging version 0.81.0-alpha-33-gb6c4fbbeb` (last MSYS2 dev build)

I used `viewport_resolution = 480x360` in the config, and this is what gets logged when using 150% display scaling in Windows 10:

```
2022-12-28 11:20:13.039 | DISPLAY: EGA 320x200 16 color (Dh) at 59.923 Hz VFR, scaled by 2.3x2.7 to 720x540 with 1.2 pixel-aspect
```

480 * 1.5 = 720 and 360 * 1.5 = 540, so that checks out; the render texture is being created with the correct dimensions.

(Ignore the `scaled by 2.3x2.7 ` part; that is wrong because we're displaying it with a single decimal digit precision... I'll fix that up in a minute)

The resulting image has the correct 720x540 pixel dimensions too (easier to measure it on the second image), and the output of the CRT shader looks absolutely correct.

I think we should retroactively add "High-DPI support on Windows" to our release notes @kcgen... Could you provide the exact wording @GranMinigun? Just to make sure we get it right this time 😅 

![image](https://user-images.githubusercontent.com/698770/209742793-53132978-b705-46f0-a290-d5c31f3782db.png)

![image](https://user-images.githubusercontent.com/698770/209742904-86b72227-cbc7-4195-a289-9fee80d526f3.png)

**Comment by GranMinigun** [comment]
Good to see it actually works. @ericwa, thanks for your assistance. As for release notes, I didn't see any reports of it working properly on multi-monitor Windows systems, which is the main point for high DPI enablement there, in my opinion (it works somewhat differently to macOS). Specifically different DPI scaling on different monitors. Something like: "Following macOS, HiDPI now comes to Windows: the window should scale nicely along other programs, and retain correct dimensions when moving between monitors with different scaling set." Just a one-liner, letting people know it's there, nothing fancy (but if you do want to go fancy, then - by all means).

**Comment by johnnovak** [comment]
>  As for release notes, I didn't see any reports of it working properly on multi-monitor Windows systems

I'm a bit confused by this statement. Did you omit a "not" from that sentence? ("I didn't see any reports of it **_not_** working properly on multi-monitor Windows systems...")

So you're saying that no one has complained about it being broken yet, therefore we should just include your above summary in the release notes, right?

**Comment by kcgen** [comment]
> > As for release notes, I didn't see any reports of it working properly on multi-monitor Windows systems
> 
> I'm a bit confused by this statement. Did you omit a "not" from that sentence? ("I didn't see any reports of it **_not_** working properly on multi-monitor Windows systems...")
> 
> So you're saying that no one has complained about it being broken yet, therefore we should just include your above summary in the release notes, right?

Oh, yes - the goal was to get proof that it's OK on dual/mutli-display setups before saying it's working.  Calls were made, but I don't think any screenshots came back. 
(GM put the call out: https://github.com/dosbox-staging/dosbox-staging/pull/2105#issuecomment-1345678736)

And myself and @kklobe chatted about it too (blurring irrelevant topics):

![2022-12-27_23-18](https://user-images.githubusercontent.com/1557255/209774051-bd55d3fe-138d-45b5-83af-92745f4841ec.png)

**Comment by GranMinigun** [comment]
> Did you omit a "not" from that sentence?

No, the statement is correct. No reports of it working correctly from users with multi-monitor systems. I asked for screenshots, but the only ones that I got, long after the merge, were from you and now from Eric. In both cases, no second display with a different DPI set. Until I see the proof, I can't confidently say "yep, it Just Works™." (Dual-display was confirmed working properly on macOS long time ago, but again, HiDPI works there differently.)

**Comment by johnnovak** [comment]
Thanks @ericwa & @GranMinigun, I'll mention this in the updated release notes.
